### PR TITLE
feat: Monorepo path triggers

### PR DESCRIPTION
**Pull request by greaka**

### CODE REVIEW

**Comment by loispostula** [comment]
@greaka I'm confused with #63, is it the same use case?

**Comment by greaka** [comment]
Yes, this branch is based on #63 and only adds support for multi-workspace monorepos.

File: .github/workflows/rust-test.yml [hunk 1]
```diff
@@ -68,20 +72,72 @@ env:
...
+          fi
+
+          git fetch origin ${{ github.base_ref }}
+
+          output_paths=''
+          root_pwd="$PWD"
+
+          for workspace in "${workspace_list[@]}"; do
+            cd "$root_pwd"
+            cd "$workspace"
```
Comment (loispostula) [inline .github/workflows/rust-test.yml:100]: Shouldn't we use `pushd` `popd` instead?

Comment (greaka) [reply .github/workflows/rust-test.yml:100]: An earlier version used them, yes. This variant is more stable against accidentally forgetting a popd when you break or continue the loop and also is easier on your shell if you run the script locally. If you feel strongly about it, then I can change it back.

**Review by loispostula — APPROVED** [review]

**Pull request merged**